homeassistant/auth/providers/external.py

```diff
@@ -0,0 +1,149 @@
+"""Auth provider that validates credentials via an external script."""
+
+import typing as T
+
+import asyncio.subprocess
+import collections
+import logging
+
+import voluptuous as vol
+
+from homeassistant.exceptions import HomeAssistantError
+
+from . import AuthProvider, AUTH_PROVIDER_SCHEMA, AUTH_PROVIDERS, LoginFlow
+from ..models import Credentials, UserMeta
+
+
+CONF_PROGRAM = "program"
+CONF_ARGS = "args"
+
+CONFIG_SCHEMA = AUTH_PROVIDER_SCHEMA.extend({
+    vol.Required(CONF_PROGRAM): str,
+    vol.Optional(CONF_ARGS, default=None): vol.Any(vol.DefaultTo(list), [str]),
+}, extra=vol.PREVENT_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InvalidAuthError(HomeAssistantError):
+    """Raised when authentication with given credentials fails."""
+
+
+@AUTH_PROVIDERS.register("external")
```

Comment:
Name is ambiguous, all auth providers besides "homeassistant" are external…

I would suggest "command_line"

Reply:
Agreed, will change that tomorrow.